### PR TITLE
Create fdc test data

### PR DESCRIPTION
**Pull request by vvartakMoJ**
## What

Added methods to allow creation of FDC test data.

The logic applied is described in MIRO flowcharts here: https://miro.com/app/board/uXjVMBupWRs=/?moveToWidget=3458764592275243441&cot=14

The class FdcTestDataCreatorServiceTest contains unit tests that verify that the expected API calls are made (using a MockWebServer with custom dispatcher).

[Link to story](https://dsdmoj.atlassian.net/browse/DCES-367, DCES-365, DCES-377 and DCES-378)

## Checklist

Before you ask people to review this PR:

- [ ] Tests should be passing: `./gradlew test`
- [ ] Github should not be reporting conflicts; you should have recently run `git rebase main`.
- [ ] Avoid mixing whitespace changes with code changes in the same commit. These make diffs harder to read and conflicts more likely.
- [ ] You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- [ ] You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/client/TestDataClient.java [hunk 1]
```diff
@@ -1,33 +1,69 @@
...
+
+    @PatchExchange("/debt-collection-enforcement/fdc-contribution")
+    @Valid
+    int updateFdcContribution(@RequestBody UpdateFdcContributionRequest fdcContribution);
+
+    @GetExchange("/assessment/rep-orders")
+    @Valid
+    List<Integer> getRepOrders(@RequestParam int delay, @RequestParam String dateReceived, @RequestParam int numRecords, @RequestParam boolean fdcDelayedPickup, @RequestParam boolean fdcFastTrack);
+
     @GetExchange("/contribution-file/{contributionFileId}")
```
Comment (dconneely) [inline dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/client/TestDataClient.java:67]: Shouldn't the /contribution-file/ clients be moved to /debt-collection-enforcement/contribution-file/ - or has maat-cd-api changed too?

File: dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/model/external/CreateFdcContributionRequest.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+package uk.gov.justice.laa.crime.dces.integration.model.external;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import io.swagger.v3.oas.annotations.media.Schema;
+import jakarta.validation.constraints.NotNull;
+import jakarta.validation.constraints.Size;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+@Data
```
Comment (dconneely) [inline dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/model/external/CreateFdcContributionRequest.java:10]: This comment is about all the classes in `uk.gov.justice.laa.crime.dces.integration.model.external`:

Are these just copied-and-pasted from maat-cd-api? They have very inconsistent lombok annotations (some use `@AllArgsConstructor`, `@NoArgsConstructor`, `@Data`, `@Builder`, etc. - not many use final fields which is one of the reasons to use `@Builder`)

File: dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/maatapi/MaatApiWebClientFactory.java [hunk 3]
```diff
@@ -52,6 +54,8 @@ public WebClient maatApiWebClient(
                     .resolver(DefaultAddressResolverGroup.INSTANCE)
                     .compress(true)
                     .responseTimeout(Duration.ofSeconds(90))
+                    .wiretap("reactor.netty.http.client.HttpClient",
+                        LogLevel.DEBUG, AdvancedByteBufFormat.TEXTUAL)
```
Comment (dconneely) [inline dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/maatapi/MaatApiWebClientFactory.java:58]: Will this produce a lot of logging? Should it maybe be enabled by a configuration property?

File: dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/client/TestDataClient.java (hunk 1 above)
Comment (vvartakMoJ) [reply dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/client/TestDataClient.java:67]: Good spot, @dconneely! This came from a merge from main and I didn't realise that the `/debt-collection-enforcement` used to be the default root of all endpoints in this class which I had to remove as part of my changes. I'll add that in the next commit.

File: dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/maatapi/MaatApiWebClientFactory.java (hunk 3 above)
Comment (vvartakMoJ) [reply dces-drc-integration/src/main/java/uk/gov/justice/laa/crime/dces/integration/maatapi/MaatApiWebClientFactory.java:58]: Yes, probably. I found it useful during debugging but it's not needed any more, so will remove it.

File: dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/model/external/CreateFdcContributionRequest.java (hunk 2 above)
Comment (vvartakMoJ) [reply dces-drc-integration/src/integrationTest/java/uk/gov/justice/laa/crime/dces/integration/model/external/CreateFdcContributionRequest.java:10]: Yes, they were originally copy-pasted from maat-cd-api. I have now removed some annotations that I could. Please check the ones in `UpdateConcorContributionStatusRequest` as they might affect your tests.

**Review by LAA-Dave-Edwards — APPROVED** [review]
Looks good to go.

**Pull request merged**